### PR TITLE
Fixing evm-debug

### PR DESCRIPTION
**Pull request by tomusdrw**

### CODE REVIEW

File: ethcore/src/evm/evm.rs [hunk 1]
```diff
@@ -113,7 +113,7 @@ impl<'a> Finalize for Result<GasLeft<'a>> {
 }
 
 /// Cost calculation type. For low-gas usage we calculate costs using usize instead of U256
-pub trait CostType: ops::Mul<Output=Self> + ops::Div<Output=Self> + ops::Add<Output=Self> + ops::Sub<Output=Self> + ops::Shr<usize, Output=Self> + ops::Shl<usize, Output=Self> + cmp::Ord + Sized + From<usize> + Copy {
+pub trait CostType: ops::Mul<Output=Self> + ops::Div<Output=Self> + ops::Add<Output=Self> + ops::Sub<Output=Self> + ops::Shr<usize, Output=Self> + ops::Shl<usize, Output=Self> + cmp::Ord + Sized + From<usize> + Copy + fmt::Debug {
```
Comment (rphmeier) [inline ethcore/src/evm/evm.rs]: this line should be broken up

File: ethcore/src/evm/interpreter/mod.rs [hunk 2]
```diff
@@ -126,7 +126,7 @@ impl<Cost: CostType> evm::Evm for Interpreter<Cost> {
 			gasometer.current_gas = gasometer.current_gas - gas_cost;
 
 			evm_debug!({
-				println!("[0x{:x}][{}(0x{:x}) Gas: {:x}\n  Gas Before: {:x}",
+				println!("[0x{:x}][{}(0x{:x}) Gas: {:?}\n  Gas Before: {:?}",
```
Comment (rphmeier) [inline ethcore/src/evm/interpreter/mod.rs]: should be one of the logging related macros (probably just `debug!`?)

Comment (tomusdrw) [inline ethcore/src/evm/interpreter/mod.rs]: It's much easier to run tests when there are printlns, cause logs are not initialized in tests. It's also a compilation feature, so when someone is compiling with it I guess you would expect the logs to be always printed.

**Comment by rphmeier** [comment]
There's a typo in the trait bounds here, needs fixing before merge.

**Comment by tomusdrw** [comment]
Shoot, I'm sorry, I haven't noticed that typo. Thanks!

**Pull request merged**